opal/corelib/array.rb

```diff
@@ -63,7 +76,15 @@ def initialize(size = nil, obj = nil, &block)
   end
 
   def self.new(*args, &block)
-    [].initialize(*args, &block)
+    obj = allocate
+    breaker = obj.initialize(*args, &block)
+    %x{
+      if (obj.$$has_breaker !== undefined) {
```

Comment:
@vais In https://github.com/opal/opal/pull/1246 you had a similar problem with `break`. Do you think it's better to make the same thing here?
``` ruby
def initialize(*)
  # ...
  %x{
    value = block(i)
    if (value === $breaker) {
      return $breaker.$v
    }
  }
end

def self.new(*)
  obj = allocate
  breaker = obj.initialize
  breaker == obj ? obj : breaker
end
```